### PR TITLE
Set window title based on type of Inspector being shown

### LINKED ISSUE #2354: Inspector window title should match name of thing it’s inspecting

### PR DESCRIPTION
**Pull request by AngeloStavrow**
This PR closes #2354.

Before showing the inspector window, this change checks what subclass of InspectorViewController class is being shown, then sets the window title accordingly:

- A `NothingInspectorViewController` (or unknown case) gets the window title "Inspector".
- A `FolderInspectorViewController`, `WebFeedInspectorViewController`, or `BuiltinSmartFeedInspectorViewController` gets the window title from the subclassed view controller's `nameTextField` property, if it exists; if it doesn't, it gets a generally-descriptive title like "Folder Inspector", "Feed Inspector", or "Smart Feed Inspector" instead.

Probably worth verifying:

- [ ] Clicking on a feed, folder, or smartfeed while the inspector is open changes the inspector's window title as expected
- [ ] Closing the inspector, clicking on another sidebar item, and then opening the inspector shows the correct window title.
- [ ] Opening the inspector and changing the name of the sidebar item (where possible) updates the inspector window's title.

👆 "Works on my machine" but please let me know if it doesn't for you!

### CODE REVIEW

**Review by brentsimmons — CHANGES_REQUESTED** [review]
Let’s take a different approach — let’s add a windowTitle var to the Inspector protocol. InspectorWindowController would then just ask the current Inspector for the windowTitle.

**Comment by AngeloStavrow** [comment]
Shoot, rebasing this branch on the merged `mac-candidate` is including commits from #2397. 😕

File: Mac/Inspector/WebFeedInspectorViewController.swift [hunk 1]
```diff
@@ -113,6 +113,7 @@ extension WebFeedInspectorViewController: NSTextFieldDelegate {
 			return
 		}
 		feed.editedName = nameTextField.stringValue
+		windowTitle = feed.editedName ?? NSLocalizedString("Feed Inspector", comment: "Feed Inspector window title")
```
Comment (AngeloStavrow) [inline Mac/Inspector/WebFeedInspectorViewController.swift:116]: @brentsimmons I'm not 100% of this WebFeedInspectorViewController change. I'm tapping into the textfield delegate's controlTextDidChange method to make sure the window title updates as the feed's name is changed. If this seems like a reasonable approach, then I think this PR is ready for re-review!

**Review by brentsimmons — CHANGES_REQUESTED** [review]
I approve those changes — but would you redo this PR? It has changes from the dock unread count issue mixed in with it.

**Comment by AngeloStavrow** [comment]
> I approve those changes — but would you redo this PR? It has changes from the dock unread count issue mixed in with it.

Yes, sorry about that! Closing this PR and replacing it with #2403.

**Pull request closed without merging**